Plugins/Animation/src/AnimationComponent.hpp

```diff
@@ -18,39 +18,87 @@ namespace AnimationPlugin {
 
+    //
+    // Build from fileData
+    //
+
+    /// Create the skeleton from the given data.
+    /// @param data: the skeleton's joint transform hierarchy.
+    /// @param duplicateTable: the map from duplicate to single vertices.
+    /// @param the number of single vertices.
+    // FIXME: the 2 last parameters are needed only for the genereation of
+    //        the skinning weights matrix.
+    void handleSkeletonLoading( const Ra::Asset::HandleData* data,
+                                const std::vector<Ra::Core::Index>& duplicateTable,
+                                uint nbMeshVertices );
+
+    /// Create the animations from the given data.
+    void handleAnimationLoading( const std::vector<Ra::Asset::AnimationData*> data );
+
+    //
+    // Animation
+    //
+
+    /// Set the animation skeleton.
     void setSkeleton( const Ra::Core::Animation::Skeleton& skel );
 
+    /// @returns the animation skeleton.
     inline Ra::Core::Animation::Skeleton& getSkeleton() { return m_skel; }
-    Ra::Core::Animation::WeightMatrix getWeights() const;
-    Ra::Core::Animation::Pose getRefPose() const;
 
     /// Update the skeleton with an animation.
     void update( Scalar dt );
+
+    /// Resets the animation, thus posing the skeleton into the reference pose.
     void reset();
-    void setXray( bool on ) const;
 
-    void toggleSkeleton( const bool status );
+    /// Saves all the state data related to the current frame into a cache file.
+    void cacheFrame( int frame ) const;
+
+    /// Restores the state data related to the \p frameID -th frame from the cache file.
+    /// \returns true if the frame has been successfully restored, false otherwise.
+    /// Note: the AnimationSystem ensures that in case the frame restoration fails,
+    ///       the Component still remains in the current frame state
+    bool restoreFrame( int frame );
+
+    /// If \p status is TRUE, then use the animation time step if available;
+    /// else, use the application timestep.
     void toggleAnimationTimeStep( const bool status );
+
+    /// Set animation speed factor.
     void setSpeed( const Scalar value );
+
+    /// Toggle the slow motion speed (speed x0.1).
     void toggleSlowMotion( const bool status );
+
+    /// Set the animation to play.
     void setAnimation( const uint i );
 
-    uint getBoneIdx( Ra::Core::Index index ) const;
+    /// @returns the current time of animation.
     Scalar getTime() const;
 
-    void handleSkeletonLoading( const Ra::Asset::HandleData* data,
-                                const std::vector<Ra::Core::Index>& duplicateTable,
-                                uint nbMeshVertices );
-    void handleAnimationLoading( const std::vector<Ra::Asset::AnimationData*> data );
+    /// @returns the duration of the current animation.
+    Scalar getDuration() const;
+
+    // Skeleton display
+    /// Turns xray display on/off for the skeleton bones.
+    void setXray( bool on ) const;
```

Comment:
Is this really related to the animation ? Shouldn't this be a property of the skeleton ?